[Release] Add daily history models and history functionality to Salesforce transform package (#45)
fivetran/dbt_salesforce: README.md

- Produces modeled tables that leverage Salesforce data from [Fivetran's connector](https://fivetran.com/docs/applications/salesforce) in the format described by [this ERD](https://fivetran.com/docs/applications/salesforce#schema) and builds off the output of our [Salesforce source package](https://github.com/fivetran/dbt_salesforce_source). 
- This package also provides you with the option to leverage the history mode to gather historical records of your essential tables.

- This package enables users to:
  - Understand the performance of your opportunities
  - Drill into how the members of your sales team are performing
  - Have a daily summary of sales activities 
  - Leverage an enhanced contact list
  - View more details about opportunity line items
  - Gather daily historical records of your accounts, contacts and opportunities

<!--section="salesforce_transformation_model"-->
This package also generates a comprehensive data dictionary of your source and modeled Salesforce data via the [dbt docs site](https://fivetran.github.io/dbt_salesforce/)
You can also refer to the table below for a detailed view of all models materialized by default within this package.

> |**model**|**description**|**Quickstart compatible**

Author: sounds good to me!

| [salesforce__account_daily_history](https://fivetran.github.io/dbt_salesforce/#!/model/model.salesforce.salesforce__account_daily_history) | Each record is a daily record in an account, starting with its first active date and updating up toward either the current date (if still active) or its last active date. | No
| [salesforce__contact_daily_history](https://fivetran.github.io/dbt_salesforce/#!/model/model.salesforce.salesforce__contact_daily_history) |  Each record is a daily record in an contact, starting with its first active date and updating up toward either the current date (if still active) or its last active date.
 | No
| [salesforce__opportunity_daily_history](https://fivetran.github.io/dbt_salesforce/#!/model/model.salesforce.salesforce__opportunity_daily_history) | Each record is a daily record in an opportunity, starting with its first active date and updating up toward either the current date (if still active) or its last active date. | No
 

> **Note**: For Quickstart Data Model users only, in addition to the above output models that are Quickstart compatible, you will also receive models in your transformation list which replicate **all** of your Salesforce objects with the inclusion of the relevant formula fields in the generated output models.

Author: Fixed.

<!--section-end-->

# 🎯 How do I use the dbt package?
## Step 1: Pre-Requisites
You will need to ensure you have the following before leveraging the dbt package.
- **Connector**: Have the Fivetran Salesforce connector syncing data into your warehouse. 
- **Database support**: This package has been tested on **BigQuery**, **Snowflake**, **Redshift**, **Databricks**, and **Postgres**. Ensure you are using one of these supported databases.

### Databricks Dispatch Configuration
If you are using a Databricks destination with this package you will need to add the below (or a variation of the below) dispatch configuration within your `dbt_project.yml`. This is required in order for the package to accurately search for macros within the `dbt-labs/spark_utils` then the `dbt-labs/dbt_utils` packages respectively.
```yml
dispatch:
  - macro_namespace: dbt_utils
    search_order: ['spark_utils', 'dbt_utils']
```

### Database Incremental Strategies 

> Some of the end models in this package are materialized incrementally. We have chosen `insert_overwrite` as the default strategy for **BigQuery** and **Databricks** databases, as it is only available for these dbt adapters. For **Snowflake**, **Redshift**, and **Postgres** databases, we have chosen `delete+insert` as the default strategy.

Reviewer: ```suggestion
The history end models in this package are materialized incrementally. We have chosen `insert_overwrite` as the default strategy for **BigQuery** and **Databricks** databases, as it is only available for these dbt adapters. For **Snowflake**, **Redshift**, and **Postgres** databases, we have chosen `delete+insert` as the default strategy.
```

Author: Committed

    salesforce_database: your_database_name    
    salesforce_schema: your_schema_name
```
### Optional: Configuring Salesforce History Mode in tandem with Salesforce
 
If you are utilizing Salesforce History Mode and your target database and schema differ as well, you will need to add an additional configuration for the history schema and database to your `dbt_project.yml`

```yml
vars:
    salesforce_database: your_database_name    
    salesforce_schema: your_schema_name

    salesforce_history_database: your_history_database_name
    salesforce_history_schema: your_history_schema_name
```

> If you wish to switch to utilizing ONLY Salesforce History Mode source tables, you should change the default config set up in your source yml files within your `dbt_project.yml`.

Author: Updated this with the new README section created in the source section, with slight modifications.

```yml
sources:
  salesforce_source: 
    salesforce:
      +enabled: false # True by default. Disable if you wish to not utilize the default Salesforce source tables.
    salesforce_history:
      +enabled: true # False by default. Enable if you wish to utilize the default Salesforce source tables.

> ``` 

Author: Removed.

```
The corresponding metrics from the disabled tables will not populate in the downstream models.

> ## Optional: Utilizing Salesforce History Mode records

Reviewer: Please see my source package review notes for similar updates that should be applied to this README as well.

Author: Updated

  salesforce__contact_history_enabled: true  # False by default. Only use if you have history mode enabled and wish to leverage the contact history table.
  salesforce__event_history_enabled: true    # False by default. Only use if you have history mode enabled and wish to leverage the event history table.
  salesforce__lead_history_enabled: true         # False by default. Only use if you have history mode enabled and wish to leverage the lead history table.
  salesforce__opportunity_history_enabled: true      # False by default. Only use if you have history mode enabled and wish to leverage the opportunity history table.
  salesforce__task_history_enabled: true         # False by default. Only use if you have history mode enabled and wish to leverage the task history table.
  salesforce__user_history_enabled: true         # False by default. Only use if you have history mode enabled and wish to leverage the user history table.
  salesforce__user_role_history_enabled: true        # False by default. Only use if you have history mode enabled and wish to leverage the user role history table.

> ```

Author: Consolidated. Also removing variables no longer being used.

### Salesforce History Mode Active Records

> If you have Salesforce [History Mode](https://fivetran.com/docs/getting-started/feature/history-mode) enabled for your connector, in your `dbt_project.yml`, you will want to add and set the desired `using_[table]_history_mode_active_records` variable(s) as `true` to filter for only active records as the standard salesforce models are designed for non-historical data. These variables are disabled by default. 

Author: This functionality has been deprecated, so excising this section.
